src/db/spanner/models.rs

```diff
@@ -123,87 +128,52 @@ impl SpannerDb {
             .hub
             .projects()
             .instances_databases_sessions_execute_sql(sql, session)
-            .doit();
-        let rv = match result {
-            Err(e) => {
-                println!("err {}", e);
-                // TODO Return error
-                Ok(0)
-            }
-            Ok(res) => {
-                println!("ok {:?}", res.1);
-                match res.1.rows {
-                    Some(row) => Ok(row[0][0].parse().unwrap()),
-                    None => Ok(0),
-                }
-            }
-        };
-        if let Ok(val) = rv {
-            self.coll_cache.put(val, name.to_owned())?;
-        };
-        rv
+            .doit()?;
+        eprintln!("ok {:?}", result.1);
+        let rows = result.1.rows.ok_or(DbErrorKind::CollectionNotFound)?;
+        let id = rows[0][0]
+            .parse::<i32>()
+            .map_err(|e| DbErrorKind::Integrity(e.to_string()))?;
+        self.coll_cache.put(id, name.to_owned())?;
+        Ok(id)
     }
 
     pub(super) fn create_collection(&self, name: &str) -> Result<i32> {
         // XXX: handle concurrent attempts at inserts
         let spanner = &self.conn;
         let session = spanner.session.name.as_ref().unwrap();
 
-        let mut sql = ExecuteSqlRequest::default();
-        sql.sql = Some("SELECT MAX(collectionid) from collections".to_string());
+        let sql = self.sql_request("SELECT COALESCE(MAX(collectionid), 1) from collections")?;
         let result = spanner
             .hub
             .projects()
             .instances_databases_sessions_execute_sql(sql, session)
-            .doit();
-        let id = match result {
-            Err(e) => {
-                println!("err {}", e);
-                // TODO Return error
-                Ok(0)
-            }
-            Ok(res) => {
-                println!("ok {:?}", res.1);
-                match res.1.rows {
-                    Some(row) => Ok(row[0][0].parse::<i32>().unwrap() + 1),
-                    None => Ok(1),
-                }
-            }
+            .doit()?;
+        let id = if let Some(rows) = result.1.rows {
+            let max = rows[0][0]
+                .parse::<i32>()
+                .map_err(|e| DbErrorKind::Integrity(e.to_string()))?;
+            max + 1
+        } else {
+            // XXX: should never happen but defaulting to id of 1 is bad
+            1
```

Comment:
Might want to put some `dbg!` or return an `Err` then. You never know, and unless you flag it, you never will.